packages/main/src/plugin/extension-loader.ts

```diff
@@ -1074,6 +1078,145 @@ export class ExtensionLoader {
+      }
+    };
+    const navigation: typeof containerDesktopAPI.navigation = {
+      navigateToContainers: async (): Promise<void> => {
+        navigateTo({ page: NavigationPage.CONTAINERS });
+      },
+      navigateToContainer: async (id: string): Promise<void> => {
+        await assertContainerIdExist(id);
+        navigateTo({
+          page: NavigationPage.CONTAINER,
+          parameters: {
+            id: id,
+          },
+        });
+      },
+      navigateToContainerLogs: async (id: string): Promise<void> => {
+        await assertContainerIdExist(id);
+        navigateTo({
+          page: NavigationPage.CONTAINER_LOGS,
+          parameters: {
+            id: id,
+          },
+        });
+      },
+      navigateToContainerInspect: async (id: string): Promise<void> => {
+        await assertContainerIdExist(id);
+        navigateTo({
+          page: NavigationPage.CONTAINER_INSPECT,
+          parameters: {
+            id: id,
+          },
+        });
+      },
+      navigateToContainerTerminal: async (id: string): Promise<void> => {
+        await assertContainerIdExist(id);
+        navigateTo({
+          page: NavigationPage.CONTAINER_TERMINAL,
+          parameters: {
+            id: id,
+          },
+        });
+      },
+      navigateToImages: async (): Promise<void> => {
+        navigateTo({
+          page: NavigationPage.IMAGES,
+        });
+      },
+      navigateToImage: async (id: string, engineId: string, tag: string): Promise<void> => {
+        await assertImageExist(id, engineId, tag);
+        navigateTo({
+          page: NavigationPage.IMAGE,
+          parameters: {
+            id: id,
+            engineId: engineId,
+            tag: tag,
+          },
+        });
+      },
+      navigateToVolumes: async (): Promise<void> => {
+        navigateTo({
+          page: NavigationPage.VOLUMES,
+        });
+      },
+      navigateToVolume: async (name: string, engineId: string): Promise<void> => {
+        await assertVolumeExist(name, engineId);
+        navigateTo({
+          page: NavigationPage.VOLUME,
+          parameters: {
+            name: name,
+            engineId: engineId,
+          },
+        });
+      },
+      navigateToPods: async (): Promise<void> => {
+        navigateTo({
+          page: NavigationPage.PODS,
+        });
+      },
+      navigateToPod: async (kind: string, name: string, engineId: string): Promise<void> => {
+        await assertPodExist(kind, name, engineId);
+        navigateTo({
+          page: NavigationPage.POD,
+          parameters: {
+            kind: kind,
+            name: name,
+            engineId: engineId,
+          },
+        });
+      },
+      navigateToContribution: async (name: string): Promise<void> => {
+        await assertContributionExist(name);
+        navigateTo({
+          page: NavigationPage.CONTRIBUTION,
+          parameters: {
+            name: name,
+          },
+        });
+      },
+    };
```

Comment:
it make me feels wondering if we should not delegate / move these lines to a dedicated file

usually we're just delegating but here you're adding logic (all assert calls)